fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -394,15 +413,27 @@ public static Map<String, Set<String>> getMVPartitionNameMapOfExternalTable(Tabl
         // Get the index of partitionColumn when table has multi partition columns.
         int partitionColumnIndex = checkAndGetPartitionColumnIndex(partitionColumns, partitionColumn);
         Map<String, Set<String>> mvPartitionKeySetMap = Maps.newHashMap();
-        for (String partitionName : partitionNames) {
-            List<String> partitionNameValues = toPartitionValues(partitionName);
-            PartitionKey partitionKey = createPartitionKey(
-                    ImmutableList.of(partitionNameValues.get(partitionColumnIndex)),
-                    ImmutableList.of(partitionColumns.get(partitionColumnIndex)),
-                    table.getType());
-            String mvPartitionName = generateMVPartitionName(partitionKey);
-            mvPartitionKeySetMap.computeIfAbsent(mvPartitionName, x -> Sets.newHashSet())
-                    .add(partitionName);
+        if (table.isJDBCTable()) {
+            for (String partitionName : partitionNames) {
+                PartitionKey partitionKey = createPartitionKey(
+                        ImmutableList.of(partitionName),
+                        ImmutableList.of(partitionColumn),
+                        table.getType());
+                String mvPartitionName = generateMVPartitionName(partitionKey);
+                mvPartitionKeySetMap.computeIfAbsent(mvPartitionName, x -> Sets.newHashSet())
+                        .add(partitionName);
+            }
+        } else {
+            for (String partitionName : partitionNames) {
+                List<String> partitionNameValues = toPartitionValues(partitionName);
```

Comment:
jdbc and hive have most same logic, put it into a function?

Reply:
After considering merging, following my approach, the code will make judgments within the loop, which is not an efficient approach. Can you provide clearer guidance?

Reply:
done